src/AutoRest.CSharp/Mgmt/Generation/SubscriptionExtensionsWriter.cs

```diff
@@ -159,18 +184,37 @@ private void WritePagingOperation(CodeWriter writer, PagingMethod pagingMethod,
             WritePagingOperationDefinition(writer, pagingMethod, async, "restOperations", "clientDiagnostics");
         }
 
-        private void WriteListResourceMethod(CodeWriter writer, Resource resource, ResourceOperation resourceOperation, PagingMethod pagingMethod, bool async)
+        private void WriteListMethod(CodeWriter writer, CSharpType pageType, MgmtRestClient restClient, PagingMethod pagingMethod, bool async, bool isResource = true)
         {
-            writer.WriteXmlDocumentationSummary($"Lists the {resource.Type.Name}s for this {typeof(SubscriptionOperations)}.");
+            writer.WriteXmlDocumentationSummary($"Lists the {pageType.Name}s for this {typeof(SubscriptionOperations)}.");
             writer.WriteXmlDocumentationParameter("subscription", $"The <see cref=\"{typeof(SubscriptionOperations)}\" /> instance the method will execute against.");
+
+            var parameters = pagingMethod.Method.Parameters;
+            foreach (var parameter in parameters)
+            {
+                writer.WriteXmlDocumentationParameter(parameter.Name, parameter.Description);
+            }
+
             writer.WriteXmlDocumentationParameter("cancellationToken", "The cancellation token to use.");
             writer.WriteXmlDocumentation("return", $"A collection of resource operations that may take multiple service requests to iterate over.");
+            writer.WriteXmlDocumentationRequiredParametersException(parameters);
 
-            var pageType = resource.Type;
             CSharpType responseType = async ? new CSharpType(typeof(AsyncPageable<>), pageType) : new CSharpType(typeof(Pageable<>), pageType);
-            var methodName = $"List{resource.Type.Name}";
-            using (writer.Scope($"public static {responseType} {CreateMethodName(methodName, async)}(this {typeof(SubscriptionOperations)} subscription, {typeof(CancellationToken)} cancellationToken = default)"))
+            var methodName = $"List{pageType.Name}";
```

Comment:
I am not sure if we could just hard-code the `List` keyword here regardless the method name defined in the corresponding `operationId`. 
But I suppose this is reasonable here, this method is only generating those method that appears to be a `List` method.

Reply:
I suppose we should `ToPlural()`, right?